whois-nrtm4-client/src/main/java/net/ripe/db/nrtm4/client/importer/DeltaImporter.java

```diff
@@ -0,0 +1,175 @@
+package net.ripe.db.nrtm4.client.importer;
+
+import io.netty.util.internal.StringUtil;
+import net.ripe.db.nrtm4.client.client.MirrorDeltaInfo;
+import net.ripe.db.nrtm4.client.client.NrtmRestClient;
+import net.ripe.db.nrtm4.client.client.UpdateNotificationFileResponse;
+import net.ripe.db.nrtm4.client.condition.Nrtm4ClientCondition;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientInfoRepository;
+import net.ripe.db.nrtm4.client.dao.Nrtm4ClientRepository;
+import net.ripe.db.nrtm4.client.dao.NrtmClientVersionInfo;
+import net.ripe.db.whois.common.dao.RpslObjectUpdateInfo;
+import net.ripe.db.whois.common.rpsl.RpslObject;
+import org.apache.commons.compress.utils.Lists;
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+@Service
+@Conditional(Nrtm4ClientCondition.class)
+public class DeltaImporter implements Importer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeltaImporter.class);
+
+    private final NrtmRestClient nrtmRestClient;
+
+    private final Nrtm4ClientRepository nrtm4ClientRepository;
+
+    private final Nrtm4ClientInfoRepository nrtm4ClientInfoRepository;
+
+
+    public static final String RECORD_SEPARATOR = "\u001E";
+
+    public DeltaImporter(final NrtmRestClient nrtmRestClient,
+                         final Nrtm4ClientRepository nrtm4ClientRepository,
+                         final Nrtm4ClientInfoRepository nrtm4ClientInfoRepository) {
+        this.nrtmRestClient = nrtmRestClient;
+        this.nrtm4ClientRepository = nrtm4ClientRepository;
+        this.nrtm4ClientInfoRepository = nrtm4ClientInfoRepository;
+    }
+
+    @Override
+    public void doImport(final String source, final UpdateNotificationFileResponse updateNotificationFile){
+        final List<UpdateNotificationFileResponse.NrtmFileLink> deltas = getNewDeltas(source, updateNotificationFile);
+
+        if (deltas.isEmpty()) {
+            LOGGER.info("No new deltas to be processed");
```

Comment:
Log this at DEBUG or we'll get constant logging whenever there are no changes

Reply:
I have almost everything to info for testing purposes.
I will turn the logs into DEBUG once finished testing